package.json

```diff
@@ -43,6 +43,7 @@
     "gulp-mocha": "^7.0.2",
     "gulp-sourcemaps": "^2.6.5",
     "gulp-typescript": "^5.0.1",
+    "kind-of": "^6.0.3",
```

Comment:
Does this achieve the intended result though? A few things I noticed:
* This line was added to `devDependencies` instead of `dependencies` (but I'm not sure if that's the reason for the following points).
* It does not seem to make a difference to `npm install`. Looking first at this repo itself:

**Before this PR:**
```
$ cd balena-io-modules/docker-qemu-transpose
$ rm -rf node_modules
$ npm install
$ npm list &> npmlist.txt
$ grep kind-of npmlist.txt | wc -l
      26
```
The 26 occurrences of `kind-of` indeed included old versions 1.1.0, 3.2.2, 4.0.0 and 5.1.0, in addition to 6.0.3. There were no occurrences of 6.0.2.

**After this PR:**
```
$ cd balena-io-modules/docker-qemu-transpose
$ rm -rf node_modules
$ npm install
$ npm list &> npmlist2.txt
$ grep kind-of npmlist2.txt | wc -l
      27
```
**diff:**
```
$ diff npmlist.txt npmlist2.txt
83c83
< │ │ │ │ ├── kind-of@6.0.3
---
> │ │ │ │ ├── kind-of@6.0.3 deduped
806a807
> ├── kind-of@6.0.3
```
So, no difference, but presumably the intention was for it to make a difference downstream, not to make a difference here.

* Looking downstream at `balena-cli`, it does not seem to make a difference either. I edited the `package-json` of `balena-cli` to make the following test:

**Before this PR, regarding the `balena-cli` repo:**
```text
# package.json
"docker-qemu-transpose": "^1.0.4",

$ rm -rf node_modules/ && npm install
$ npm list &> npmlist.txt
$ grep kind-of npmlist.txt | wc -l
      28
$ grep kind-of\@6.0.2 npmlist.txt | wc -l
      18
```
So indeed there were 18 occurrences of `kind-of\@6.0.2`.

**After this PR, regarding the `balena-cli` repo:**
```text
# package.json
"docker-qemu-transpose": "^1.0.5-update-kind-of-73d02c3d6a93419b52254e33e103de0c91055829",

$ rm -rf node_modules/ && npm install

$ grep 6.0.3 node_modules/docker-qemu-transpose/package.json
    "kind-of": "^6.0.3",

$ npm list &> npmlist2.txt
$ grep kind-of npmlist2.txt | wc -l
      29
$ grep kind-of\@6.0.2 npmlist2.txt | wc -l
      18
$ grep kind-of\@6.0.3 npmlist2.txt | wc -l
       0
$ diff npmlist.txt npmlist2.txt
1219c1219
< ├─┬ docker-qemu-transpose@1.0.4
---
> ├─┬ docker-qemu-transpose@1.0.5-update-kind-of-73d02c3d6a93419b52254e33e103de0c91055829
```

So, it made not difference to `balena-cli`, even though I confirmed (with `git diff`) that the `npm-shrinkwrap.json` file of balena-cli was updated as intended:

```
diff --git a/npm-shrinkwrap.json b/npm-shrinkwrap.json
index c01c53f7..e296c02f 100644
--- a/npm-shrinkwrap.json
+++ b/npm-shrinkwrap.json
@@ -4514,9 +4514,9 @@
       }
     },
     "docker-qemu-transpose": {
-      "version": "1.0.4",
-      "resolved": "https://registry.npmjs.org/docker-qemu-transpose/-/docker-qemu-transpose-1.0.4.tgz",
-      "integrity": "sha512-sSg13dXH3cRZ/svJc1zigMcgPuKm85Cjx+Eg6bKYumg1mUTIeiIsacK79sHxoxk8r4keVDfvQ+/8507HUJac3w==",
+      "version": "1.0.5-update-kind-of-73d02c3d6a93419b52254e33e103de0c91055829",
+      "resolved": "https://registry.npmjs.org/docker-qemu-transpose/-/docker-qemu-transpose-1.0.5-update-kind-of-73d02c3d6a93419b52254e33e103de0c91055829.tgz",
+      "integrity": "sha512-fbmgOaAIWFKHFsNogsC5dkItQXVYEezXQp2BQPpxGUw8WXt8FUo0gWZIheanCDLzOqw9DZh0KhuIpVLdjUhRcw==",
       "requires": {
         "@types/bluebird": "^3.5.30",
         "@types/event-stream": "^3.3.34",
diff --git a/package.json b/package.json
index 99f664b1..4ebcdf74 100644
--- a/package.json
+++ b/package.json
@@ -190,7 +190,7 @@
     "denymount": "^2.3.0",
     "docker-modem": "^2.1.1",
     "docker-progress": "^4.0.1",
-    "docker-qemu-transpose": "^1.0.4",
+    "docker-qemu-transpose": "^1.0.5-update-kind-of-73d02c3d6a93419b52254e33e103de0c91055829",
     "docker-toolbelt": "^3.3.8",
     "dockerode": "^2.5.8",
     "ejs": "^3.0.1",
```

* The source code of `docker-qemu-transpose` itself does not make any use of `kind-of`.

I am not sure what we should do instead... I am not offering a solution in this comment, I am just pointing out that the proposed change to `docker-qemu-transpose` may not achieve what was intended. We could try moving `kind-of` from `devDependencies` to `dependencies`, although my test with `balena-cli` was using `npm install` on a repo folder, rather than `npm install --production`.

Reply:
The following `balena-cli` PR makes a difference, but of course it is limited to the `balena-cli` repo: https://github.com/balena-io/balena-cli/pull/1692#discussion_r402299747

Reply:
I was going to suggest we tried moving the line `"kind-of": "^6.0.3",` from `devDependencies` to `dependencies`, but it looks like all uses of `kind-of` are derived from `gulp`, which is a dev dependency...